File: 1798-max-number-of-k-sum-pairs/1798-max-number-of-k-sum-pairs.py
class Solution:
    def maxOperations(self, nums: List[int], k: int) -> int:
            import collections
            Out = 0
            C = Counter(nums)
            for j,v in C.items():
                k2 = k - j
                if k2 != k:
                    Out += min(C[j],C[k2])
                else:
                    Out += C[j]//2
            return int(Out/2)
                
            
        

File: 1798-max-number-of-k-sum-pairs/1798-max-number-of-k-sum-pairs.cpp
class Solution {
public:
#include <unordered_map>
    int maxOperations(vector<int>& nums, int k) {
        int countx;
        int x;
        int k_x;
        int countk_x;
        int Out = 0;
        std::unordered_map<int, int> frequencyMap;
        for (int num : nums) {
            frequencyMap[num]++;
        }
        for (auto s:frequencyMap) {
            x = s.first;
            k_x = k-x;
            countx = frequencyMap[x];
            if (x==k_x){Out+=countx;}
            else{
                if (frequencyMap.find(k_x) != frequencyMap.end()) {
                    countk_x = frequencyMap[k_x];
                    Out+=min(countx,countk_x);
                }
            }
            }
        return Out/2;
    }
};